TestPeerConnection() fails irregularly

When running tests for the peer package, `TestPeerConnection` sometimes fails once every 15-20 runs.

```
$ cd .../github.com/decred/dcrd/peer
$ go test
--- FAIL: TestPeerConnection (0.00s)
	peer_test.go:300: Running 2 tests
	peer_test.go:198: testPeer: wrong LastSend - got 2017-09-18 16:19:43 +0200 CEST, want 1970-01-01 01:00:00 +0100 CET
FAIL
exit status 1
FAIL	github.com/decred/dcrd/peer	0.040s

```

--- a/peer/peer_test.go
+++ b/peer/peer_test.go
@@ -11,6 +11,7 @@ import (
 	"io"
 	"net"
 	"strconv"
+	"sync"
 	"testing"
 	"time"
 
@@ -207,6 +208,7 @@ func testPeer(t *testing.T, p *peer.Peer, s peerStats) {
 
 // TestPeerConnection tests connection between inbound and outbound peers.
 func TestPeerConnection(t *testing.T) {
+	var pause sync.Mutex
 	verack := make(chan struct{})
 	peerCfg := &peer.Config{
 		Listeners: peer.MessageListeners{
@@ -218,6 +220,8 @@ func TestPeerConnection(t *testing.T) {
 				if _, ok := msg.(*wire.MsgVerAck); ok {
 					verack <- struct{}{}
 				}
+				pause.Lock()
+				pause.Unlock()
 			},
 		},
 		UserAgentName:    "peer",
@@ -303,8 +307,11 @@ func TestPeerConnection(t *testing.T) {
 			t.Errorf("TestPeerConnection setup #%d: unexpected err %v", i, err)
 			return
 		}
+
+		pause.Lock()
 		testPeer(t, inPeer, wantStats)
 		testPeer(t, outPeer, wantStats)
+		pause.Unlock()
 
 		inPeer.Disconnect()
 		outPeer.Disconnect()

